pkg/cmd/step/create/step_create_values.go

```diff
@@ -194,3 +203,66 @@ func (o *StepCreateValuesOptions) CreateValuesFile() error {
 	}
 	return nil
 }
+
+func (o *StepCreateValuesOptions) templateSchemaFile(schemaFileName string, requirements *config.RequirementsConfig) error {
+	templateFile := strings.TrimSuffix(schemaFileName, ".schema.json") + ".tmpl.schema.json"
+	exists, err := util.FileExists(templateFile)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+	if requirements.Cluster.GitKind == "" {
+		requirements.Cluster.GitKind = "github"
+	}
+	if requirements.Cluster.GitServer == "" {
+		switch requirements.Cluster.GitKind {
+		case "bitbucketcloud":
+			requirements.Cluster.GitServer = "https://bitbucket.org"
+		case "github":
+			requirements.Cluster.GitServer = "https://github.com"
+		case "gitlab":
+			requirements.Cluster.GitServer = "https://gitlab.com"
+		}
```

Comment:
Is there a way to not do this in code so that we can keep the step generic and not tied to the schema?

Reply:
its more a bit of validation of the requirements object to ensure we populate git server URLs for the github/gitlab host names if none exists in the `jx-requirements.yml` - not sure how else we could do this really

Reply:
Not sure there is a good way around it tbh.